test/performance/odp_bench_buffer.c

```diff
@@ -0,0 +1,883 @@
+	if (shm == ODP_SHM_INVALID) {
+		ODPH_ERR("Error: shared mem reserve failed\n");
+		exit(EXIT_FAILURE);
+	}
+
+	gbl_args = odp_shm_addr(shm);
+	if (gbl_args == NULL) {
+		ODPH_ERR("Error: shared mem alloc failed\n");
+		exit(EXIT_FAILURE);
+	}
+
+	memset(gbl_args, 0, sizeof(args_t));
+	odp_atomic_init_u32(&gbl_args->exit_thread, 0);
+
+	gbl_args->bench = test_suite;
+	gbl_args->num_bench = sizeof(test_suite) / sizeof(test_suite[0]);
+
+	/* Parse and store the application arguments */
+	parse_args(argc, argv, &gbl_args->appl);
+
+	/* Get default worker cpumask */
+	if (odp_cpumask_default_worker(&cpumask, 1) != 1) {
+		ODPH_ERR("Error: unable to allocate worker thread\n");
+		exit(EXIT_FAILURE);
+	}
+	(void)odp_cpumask_to_str(&cpumask, gbl_args->cpumaskstr, sizeof(gbl_args->cpumaskstr));
+
+	if (odp_schedule_capability(&sched_capa)) {
+		ODPH_ERR("Error: schedule capability failed\n");
+		exit(EXIT_FAILURE);
+	}
+
+	gbl_args->max_flow_id = 0;
+	if (sched_capa.max_flow_id) {
+		odp_schedule_config_t sched_config;
+
+		odp_schedule_config_init(&sched_config);
+		sched_config.max_flow_id = 1;
+
+		if (odp_schedule_config(&sched_config)) {
+			ODPH_ERR("Error: schedule config failed\n");
+			exit(EXIT_FAILURE);
+		}
+		gbl_args->max_flow_id = 1;
+	}
+
+	if (odp_pool_capability(&capa)) {
+		ODPH_ERR("Error: unable to query pool capability\n");
+		exit(EXIT_FAILURE);
+	}
+
+	buf_num = gbl_args->appl.burst_size * TEST_REPEAT_COUNT;
+
+	if (capa.buf.max_num && capa.buf.max_num < buf_num) {
+		ODPH_ERR("Error: pool size not supported (max %" PRIu32 ")\n", capa.buf.max_num);
+		exit(EXIT_FAILURE);
+	} else if (gbl_args->appl.cache_size > (int)capa.buf.max_cache_size) {
+		ODPH_ERR("Error: cache size not supported (max %" PRIu32 ")\n",
+			 capa.buf.max_cache_size);
+		exit(EXIT_FAILURE);
+	}
+
+	gbl_args->buf_size = TEST_BUF_SIZE;
+	if (capa.buf.max_size && capa.buf.max_size < TEST_BUF_SIZE)
+		gbl_args->buf_size = capa.buf.max_size;
+
+	gbl_args->uarea_size = TEST_UAREA_SIZE < capa.buf.max_uarea_size ?
+				TEST_UAREA_SIZE : capa.buf.max_uarea_size;
+
+	print_info();
+
+	/* Create buffer pool */
+	odp_pool_param_init(&params);
+	params.buf.size = gbl_args->buf_size;
+	params.buf.num = buf_num;
+	params.buf.uarea_size = gbl_args->uarea_size;
+	if (gbl_args->appl.cache_size >= 0)
+		params.buf.cache_size = gbl_args->appl.cache_size;
+	params.type = ODP_POOL_BUFFER;
+
+	gbl_args->pool = odp_pool_create("microbench", &params);
+	if (gbl_args->pool == ODP_POOL_INVALID) {
+		ODPH_ERR("Error: pool create failed\n");
+		exit(EXIT_FAILURE);
+	}
+
+	odp_pool_print(gbl_args->pool);
+
+	memset(&worker_thread, 0, sizeof(odph_thread_t));
+
+	signal(SIGINT, sig_handler);
+
+	/* Create worker thread */
+	cpu = odp_cpumask_first(&cpumask);
+
+	odp_cpumask_zero(&thd_mask);
+	odp_cpumask_set(&thd_mask, cpu);
+
+	odph_thread_common_param_init(&thr_common);
+	thr_common.instance = instance;
+	thr_common.cpumask = &thd_mask;
```

Comment:
thd_mask is misleading name and could be renamed as cpumask. Current cpumask could be renamed to e.g. default_mask, or similar.